### PR TITLE
Fix unused function warning for decodeMIDR().

### LINKED ISSUE #12052: Universal build (x86_64 and arm64) error for MacOS
**Describe the bug**
Compilation error when building ONNX Runtime with CMake option `CMAKE_OSX_ARCHITECTURES="arm64;x86_64"`.

**Urgency**
I need such runtime for a deadline in at most 2 weeks.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS M1 12.4 Monterey
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: commit fdf12a5c3516a7a28fb2f40a446a65b5e7f0acc5
- Python version: No 
- GCC/Compiler version (if compiling from source): Apple clang version 13.1.6 (Target: arm64-apple-darwin21.5.0)

**To Reproduce**
Update the `cmake/external/abseil-cpp.cmake` file to update the abseil version:
```
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

include(FetchContent)

# Pass to build
set(ABSL_PROPAGATE_CXX_STD 1)
set(BUILD_TESTING 0)

#if(Patch_FOUND)
#  set(ABSL_PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/abseil/Fix_Nvidia_Build_Break.patch)
#else()
#  set(ABSL_PATCH_COMMAND git apply --ignore-space-change --ignore-whitespace ${PROJECT_SOURCE_DIR}/patches/abseil/Fix_Nvidia_Build_Break.patch)
#endif()

FetchContent_Declare(
    abseil_cpp
    PREFIX "${CMAKE_CURRENT_BINARY_DIR}/abseil-cpp"
    BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/external/abseil-cpp"
    URL https://github.com/abseil/abseil-cpp/archive/refs/tags/20220623.rc1.zip
    URL_HASH SHA1=3ffed603630b3da4c9eb3133799cb56040fc601e
    PATCH_COMMAND ${ABSL_PATCH_COMMAND}
)

FetchContent_MakeAvailable(abseil_cpp)
FetchContent_GetProperties(abseil_cpp SOURCE_DIR)

if (GDK_PLATFORM)
  # Abseil considers any partition that is NOT in the WINAPI_PARTITION_APP a viable platform
  # for Win32 symbolize code (which depends on dbghelp.lib); this logic should really be flipped
  # to only include partitions that are known to support it (e.g. DESKTOP). As a workaround we
  # tell Abseil to pretend we're building an APP.
  target_compile_definitions(absl_symbolize PRIVATE WINAPI_FAMILY=WINAPI_FAMILY_DESKTOP_APP)
endif()
```
In order to properly build Protobuf in such universal context. Then run the following build command:
```
/build.sh  --build_dir build --use_coreml --cmake_extra_defines CMAKE_OSX_ARCHITECTURES="arm64;x86_64"  --skip_tests  --parallel --build_shared_lib --config Release
```

The error I get after a while (at 45%) is:
```
In file included from /Users/jplu/dev/onnxruntime/onnxruntime/core/common/cpuid_info.cc:44:
/Users/jplu/dev/onnxruntime/onnxruntime/core/common/cpuid_uarch.h:202:13: error: unused function 'decodeMIDR' [-Werror,-Wunused-function]
static void decodeMIDR(
            ^
1 error generated.
make[2]: *** [CMakeFiles/onnxruntime_common.dir/Users/jplu/dev/onnxruntime/onnxruntime/core/common/cpuid_info.cc.o] Error 1
make[1]: *** [CMakeFiles/onnxruntime_common.dir/all] Error 2
```

I tried to add `--cmake_extra_defines CMAKE_CXX_FLAGS=-Wno-unused-function` in the build command line but without any success.

Any idea how to by pass this `unused function` error?

Thanks in advance.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Fix unused function warning for decodeMIDR().
Changed from static function defined in header to function declared in header and defined in separate .cc file.

**Motivation and Context**
Fix #12052

### CODE REVIEW

File: onnxruntime/core/common/cpuid_uarch.cc [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+#include "core/common/cpuid_uarch.h"
+
+#include "core/common/common.h"
+
+namespace onnxruntime {
+
+#if defined(CPUIDINFO_ARCH_ARM)
+
+#define CPUINFO_ARM_MIDR_IMPLEMENTER_MASK UINT32_C(0xFF000000)
```
Comment (edgchen1) [inline onnxruntime/core/common/cpuid_uarch.cc:12]: most of this file was moved from cpuid_uarch.h

**Comment by snnn** [comment]
How often these functions need to be called? If often,  this change could hurt performance. 
I think the root cause is "inline static". static functions should not be in header files. If you remove "static", it would be fine.

**Review by snnn — DISMISSED** [review]

**Comment by edgchen1** [comment]
@snnn I made some fixes after testing locally on Windows with the `--arm64` build.py option. However, I didn't see any CI test failures. Should we also have a CI build with this configuration?

**Comment by snnn** [comment]
> @snnn I made some fixes after testing locally on Windows with the `--arm64` build.py option. However, I didn't see any CI test failures. Should we also have a CI build with this configuration?

Pranav says we will consider it after the release.

**Review by snnn — APPROVED** [review]

**Pull request merged**